### PR TITLE
added stats mesos support code

### PR DESCRIPTION
**Pull request by kudva**
Signed-off-by: Prabhakar Kudva <kudva@us.ibm.com>

### CODE REVIEW

File: crawler/crawler_mesos.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        result = json.loads(urllib2.urlopen(CONFIGS[0]['mesos_url'], timeout=10).read())
+    except urllib2.URLError, e:
+        print "url error"
+        return None
+    print "mesos_stats" 
+    print result
+    return result
+
+
+def setup_logger(logger_name, logfile='crawler.log', process_id=None):
```
Comment (ricarkol) [inline crawler/crawler_mesos.py]: Don't have to set it up again. You should be able the crawler.log setup at crawler.py.

**Comment by ricarkol** [comment]
Prabhakar, looks good. But need to double check with the checks again after a rebase. Can you rebase your branch to master please? also, can you add a test please? you can mock the mesos server by mocking urllib2, like this: http://stackoverflow.com/questions/19203627/mocking-urllib2-urlopen-read-for-different-responses

**Comment by kudva** [comment]
I can write a test with Mock, but all your crawler tests are bash shell based right? Okay to call a python file in the test_mesos.sh?  If so, I can have it done tomorrow

**Comment by ricarkol** [comment]
I'm in the process of writing unit tests (<1 sec execution time by mocking everything) here: https://github.com/cloudviz/agentless-system-crawler/tree/master/tests/unit

**Comment by kudva** [comment]
Cool, I like that. I'll add one

File: tests/unit/test_mesos_url.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# test.py
+from mock import patch, Mock
+from config_and_metrics_crawler.crawler_mesos import fetch_stats
+from config_and_metrics_crawler.crawler_mesos import CONFIGS
```
Comment (MHBauer) [inline tests/unit/test_mesos_url.py]: `config_and_metrics_crawler.crawler_mesos` needs to be `crawler.crawler_mesos`

File: crawler/crawler_mesos.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
+#! /usr/bin/python
```
Comment (MHBauer) [inline crawler/crawler_mesos.py]: this file could be called `mesos.py` we already know we're in the crawler.

**Comment by ricarkol** [comment]
You don't have to create a new branch. If you rebase or something on this one, you can force push: `git push -f`

**Comment by MHBauer** [comment]
looking good. This can all be squashed down into one. We don't need to see the progress of fixing all the typos and file renames.

`git rebase -i 2e6af83fcdf1b993066dc600b10968dafffd8792` should get you started as the code is right now.

**Comment by MHBauer** [comment]
Squash looks good.

Code still good @ricarkol?

**Comment by ricarkol** [comment]
@MHBauer was trying to fix this with Prabhakar (we share the same building). But it got so messy after all the rebases, merges and some other unexplained weird stuff, that Prabhakar decided to create a fresh branch and try again. So, closing this PR.

**Pull request closed without merging**